src/App.js

```diff
@@ -399,13 +416,26 @@ class App extends Component {
     const menuIconSelectedStyle = {
       fill: this.state.theme.menuIconSelected
     };
-	
+    const allLayouts = this.getLayoutsList();
+    const nbLayouts = allLayouts.length;
+    const currentLayout = allLayouts.indexOf("layout" + this.configId);
+    const prevLayout = currentLayout > 0 ? currentLayout-1 : nbLayouts-1;
+    const nextLayout = currentLayout < nbLayouts-1 ? currentLayout+1 : 0;
+    const prevConfigId = nbLayouts > 1 ? allLayouts[prevLayout].substring(6) : '';
+    const nextConfigId = nbLayouts > 1 ? allLayouts[nextLayout].substring(6) : '';
+    
```

Comment:
It's not a lot of computation but it really isn't necessary to do this on each render. You could move all the above out of the render loop, in a separate function that you'd only call in handleConfigIdChange (and in componentWillMount probably), and then store your nbLayouts, prevConfigId and nextConfigId in the app state. And nbLayouts will never change in the app lifecycle (at least for now, until we add a + button or something) so your only need to compute it once at startup.